Project2015To2017.Core/Writing/ProjectWriter.cs

```diff
@@ -14,30 +14,33 @@ public sealed class ProjectWriter
 		private readonly ILogger logger;
 		private readonly Action<FileSystemInfo> deleteFileOperation;
 		private readonly Action<FileSystemInfo> checkoutOperation;
+		private readonly bool makeBackups;
+
 
 		public ProjectWriter(ILogger logger = null)
-			: this(logger, _ => { }, _ => { })
+			: this(logger, new ProjectWriteOptions {DeleteFileOperation = _ => { }})
 		{
 		}
 
-		public ProjectWriter(Action<FileSystemInfo> deleteFileOperation, Action<FileSystemInfo> checkoutOperation)
-			: this(null, deleteFileOperation, checkoutOperation)
+		public ProjectWriter(ProjectWriteOptions options)
+			: this(null, options)
 		{
 
 		}
 
-		public ProjectWriter(ILogger logger, Action<FileSystemInfo> deleteFileOperation, Action<FileSystemInfo> checkoutOperation)
+		public ProjectWriter(ILogger logger, ProjectWriteOptions options)
 		{
 			this.logger = logger ?? NoopLogger.Instance;
-			this.deleteFileOperation = deleteFileOperation;
-			this.checkoutOperation = checkoutOperation;
+			deleteFileOperation = options.DeleteFileOperation;
+			checkoutOperation = options.CheckoutOperation;
+			makeBackups = options.MakeBackups;
 		}
 
-		public bool TryWrite(Project project, bool makeBackups)
+		public bool TryWrite(Project project)
```

Comment:
Note, that since this is a relatively new API, there is no need to duplicate it with old signatures (unlike the constructors).